crates/sui-graphql-rpc/src/server/sui_sdk_data_provider.rs

```diff
@@ -165,7 +164,7 @@ impl DataProvider for SuiClient {
 
         connection.edges.extend(pg.data.into_iter().map(|n| {
             let g = n.data.unwrap();
-            let o = convert_obj(&g);
+            let o = (&g).into();
```

Comment:
nit: The pattern of calling `into` on a reference feels a bit strange/clunky.  Can we use the `from` function instead here? e.g. `Object::from(&g)`?

I think this comment applies in a couple of other places too.